### ISSUE TITLE
A* searcher could be run asynchronously until the end of turn 1 at least

### ISSUE THREAD

**Issue by SteveDraper** [SanchoGGP/ggp-base#134]
Currently we attempt to solve using the A\* searcher for puzzles with known target states during meta-gaming, but we could let it continue until we have to return the first move (just run it in parallel with MCTS during turn 1), which would give it longer to attempt the solve.

Low priority though as the only known case we use it for currently is 8-puzzle, which it solves in under a second.
